cdk/src/constructs/enable-controltower.ts

```diff
@@ -0,0 +1,108 @@
+import * as path from 'path';
+import * as pythonLambda from '@aws-cdk/aws-lambda-python-alpha';
+import { aws_lambda as lambda, CfnCustomResource, CustomResource, Stack } from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cr from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+
+
+interface EnableControltowerProps {
+  /**
+   * Email of the audit aws account
+   */
+  readonly auditAwsAccountEmail: string;
+  /**
+   * Email of the log archive aws account
+   */
+  readonly logArchiveAwsAccountEmail: string;
+}
+
+export class EnableControltower extends Construct {
+  constructor(scope: Construct, id: string, props: EnableControltowerProps) {
+    super(scope, id);
+
+    const resource = new CustomResource(this, 'Resource', {
+      serviceToken: EnableControltowerProvider.getOrCreate(this),
```

Comment:
I like getInstance more since its more obvious that its a singleton
```suggestion 
      serviceToken: EnableControltowerProvider. getInstance(this),
```